### PR TITLE
[Feature] Support schema change primary key model reorder.

### LINKED ISSUE #12684: [Feature] Support schema change primary key model sort key.
## Feature request

do not support assign sort key of primary key model by schema change reorder now.

**Is your feature request related to a problem? Please describe.**
```
MySQL [test]> CREATE TABLE test (
    ->   `k1` int,
    ->   `k2` int,
    ->   `k3` int,
    ->   `v1` int,
    ->   `v2` int,
    ->   `v3` int,
    ->   `v4` int,
    ->   `v5` int
    -> ) ENGINE=OLAP
    -> PRIMARY KEY(`k1`, `k2`, `k3`)
    -> COMMENT "OLAP"
    -> DISTRIBUTED BY HASH(`k1`, `k2`, `k3`) BUCKETS 1
    -> ORDER BY(`v2`, `v3`)
    -> PROPERTIES ("replication_num" = "1");
Query OK, 0 rows affected (0.11 sec)

MySQL [test]> ALTER TABLE test ORDER BY (v3, v2);
Primary key table do not support reorder column

MySQL [test]>
```

**Describe the solution you'd like**
```
MySQL [test]> select * from test;
+------+------+------+------+------+------+------+------+
| k1   | k2   | k3   | v1   | v2   | v3   | v4   | v5   |
+------+------+------+------+------+------+------+------+
|    1 |    5 |    1 |    5 |    1 |    5 |    1 |    5 |
|    2 |    4 |    2 |    4 |    2 |    4 |    2 |    4 |
|    3 |    3 |    3 |    3 |    3 |    3 |    3 |    3 |
|    4 |    2 |    4 |    2 |    4 |    2 |    4 |    2 |
|    5 |    1 |    5 |    1 |    5 |    1 |    5 |    1 |
+------+------+------+------+------+------+------+------+
5 rows in set (0.02 sec)

MySQL [test]> ALTER TABLE test ORDER BY (v3, v2);
Query OK, 0 rows affected (0.03 sec)

MySQL [test]> select * from test;
+------+------+------+------+------+------+------+------+
| k1   | k2   | k3   | v1   | v2   | v3   | v4   | v5   |
+------+------+------+------+------+------+------+------+
|    5 |    1 |    5 |    1 |    5 |    1 |    5 |    1 |
|    4 |    2 |    4 |    2 |    4 |    2 |    4 |    2 |
|    3 |    3 |    3 |    3 |    3 |    3 |    3 |    3 |
|    2 |    4 |    2 |    4 |    2 |    4 |    2 |    4 |
|    1 |    5 |    1 |    5 |    1 |    5 |    1 |    5 |
+------+------+------+------+------+------+------+------+
5 rows in set (0.02 sec)

MySQL [test]>
```

**Describe alternatives you've considered**
add reorder by sort key in schema change procedure.

### PR DESCRIPTION
**Pull request by Linkerist**
## What type of PR is this：
- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #12684

## Problem Summary(Required) ：

now schema change primary key model reorder is not ok.
```
MySQL [test]> ALTER TABLE test ORDER BY (k1, k2, k3, v1, v3, v2, v4, v5);
Primary key table do not support reorder column
```

this PR support it, can reorder sort key:
```
MySQL [test]> ALTER TABLE test ORDER BY (k1, k2, k3, v1, v3, v2, v4, v5);
Query OK, 0 rows affected (0.03 sec)
```

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 0 / 4 (00.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/alter/SchemaChangeHandler.java|0|4|00.00%|[483, 495, 496, 497]|

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

File: fe/fe-core/src/main/java/com/starrocks/alter/SchemaChangeHandler.java [hunk 1]
```diff
@@ -491,6 +492,10 @@ private void processReorderColumn(ReorderColumnsClause alterClause, OlapTable ol
             if (!colNameSet.add(colName)) {
                 throw new DdlException("Duplicated column[" + colName + "]");
             }
+            if (oneCol.get().isKey() && keyIdx++ != targetIndexSchema.indexOf(oneCol.get())
```
Comment (chaoyli) [inline fe/fe-core/src/main/java/com/starrocks/alter/SchemaChangeHandler.java:495]: 1. create table A(k1,k2,k3,v1,v2) primary key(k1, k2, k3) order by (k1, k2,k3)
    alter table A order by (v1,v2)
2. create table A primary key(k1, k2, k3,v1,v2) order by (v1, v2)
    alter table A order by (v2, v1) 
3. create table A primary key(k1, k2, k3,v1,v2) order by (k1, k2, k3)
    alter table A order by (k3, k2, k1)

Comment (chaoyli) [reply fe/fe-core/src/main/java/com/starrocks/alter/SchemaChangeHandler.java:495]: 1/2 is ok, 3 is not ok?

File: be/src/storage/schema_change.cpp [hunk 2]
```diff
@@ -262,15 +270,13 @@ bool ChunkMerger::_pop_heap() {
...
+    if (++element.row_index < element.chunk->num_rows()) {
+        _heap.push(element);
     }
-
-    _heap.push(element);
     return true;
 }
 
-bool LinkedSchemaChange::process(TabletReader* reader, RowsetWriter* new_rowset_writer, TabletSharedPtr new_tablet,
+bool LinkedSchemaChange::process(ChunkIterator* iter, RowsetWriter* new_rowset_writer, TabletSharedPtr new_tablet,
```
Comment (chaoyli) [inline be/src/storage/schema_change.cpp:279]: Schema change receive TabletReader may be better. The ChunkIterator is transparent to schema change class

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 0 / 4 (00.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/alter/SchemaChangeHandler.java|0|4|00.00%|[483, 495, 496, 497]|

**Comment by dulong41** [comment]
run starrocks_admit_test

**Pull request closed without merging**